core/server/master/src/main/java/alluxio/master/file/meta/InodeUtils.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.file.meta;
+
+import alluxio.AlluxioURI;
+import alluxio.exception.ExceptionMessage;
+import alluxio.master.journal.JournalContext;
+import alluxio.proto.journal.File;
+import alluxio.proto.journal.Journal;
+import alluxio.retry.ExponentialBackoffRetry;
+import alluxio.retry.RetryPolicy;
+import alluxio.security.authorization.Mode;
+import alluxio.underfs.UnderFileSystem;
+import alluxio.underfs.options.MkdirsOptions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Represents the tree of Inode's.
+ */
+public final class InodeUtils {
+  private static final Logger LOG = LoggerFactory.getLogger(InodeUtils.class);
+
+  private InodeUtils() {} // prevent instantiation
+
+  /**
+   * Synchronously persists an {@link InodeDirectory} to the UFS. If concurrent calls are made, only
+   * one thread will persist to UFS, and the others will wait until it is persisted.
+   *
+   * @param dir the {@link InodeDirectory} to persist
+   * @param inodeTree the {@link InodeTree}
+   * @param mountTable the {@link MountTable}
+   * @param journalContext the journal context
+   * @throws IOException if the file fails to persist
+   */
+  public static void syncPersistDirectory(InodeDirectory dir, InodeTree inodeTree,
+      MountTable mountTable, JournalContext journalContext) throws IOException {
+    RetryPolicy retry = new ExponentialBackoffRetry(2, 1000, 200);
```

Comment:
I wanted to wait long enough for slow UFSes. I reduced it to 50 retries, which would be between 40-50 seconds.